SerialPrograms/Source/PokemonSV/Programs/Sandwiches/PokemonSV_IngredientSession.cpp

```diff
@@ -31,16 +34,16 @@ IngredientSession::IngredientSession(
     , m_context(context)
     , m_language(language)
     , m_overlays(console.overlay())
-    , m_ingredients(10)
+    , m_type(type)
+    , m_num_confirmed(0)
     , m_arrow(COLOR_CYAN, GradientArrowType::RIGHT, {0.02, 0.15, 0.05, 0.80})
 {
-    for (size_t c = 0; c < INGREDIENT_PAGE_LINES; c++){
-        m_ingredients.emplace_back(type, c, COLOR_CYAN);
-        m_ingredients.back().make_overlays(m_overlays);
-    }
+    SandwichIngredientReader reader(m_type, COLOR_CYAN);
+    reader.make_overlays(m_overlays);

```

Comment:
I don't think this works the way you think it does. Once `reader` is destructed, it removes all the boxes from the overlay.

Reply:
I just tested it. And the boxes still show up on screen. (and the boxes go away if I remove that code)
Why does my code work?

Reply:
Looks like I confused `VideoOverlaySet` with the `VideoOverlay` itself. The latter holds references to the boxes while the former owns the boxes. So when `VideoOverlaySet` is destructed, it automatically removes the boxes it holds from the `VideoOverlay`.

You're good.